wsgi.py

```diff
@@ -6,255 +6,348 @@
+@application.route("/projects/<project_uuid>", methods=["DELETE"])
+@auth.login_required
+def delete_moc_project(project_uuid):
+    shift = get_openshift()
+    if shift.project_exists(project_uuid):
+        r = shift.delete_project(project_uuid)
+        if r.status_code == 200 or r.status_code == 201:
             return Response(
-                response=json.dumps({"msg": "project deleted (" + project_uuid +")" }),
+                response=json.dumps({"msg": "project deleted (" + project_uuid + ")"}),
                 status=200,
-                mimetype='application/json'
-               )
+                mimetype="application/json",
+            )
         return Response(
-            response=json.dumps({"msg": "project unabled to be deleted (" + project_uuid +")" }),
+            response=json.dumps(
+                {"msg": "project unabled to be deleted (" + project_uuid + ")"}
+            ),
             status=400,
-            mimetype='application/json'
-            )
+            mimetype="application/json",
+        )
     return Response(
-        response=json.dumps({"msg": "unable to delete, project does not exist(" + project_uuid +")" }),
+        response=json.dumps(
+            {"msg": "unable to delete, project does not exist(" + project_uuid + ")"}
+        ),
         status=400,
-        mimetype='application/json'
-        )
+        mimetype="application/json",
+    )
 
-@application.route("/users/<user_name>", methods=['GET'])
-def get_moc_user(user_name, full_name=None, id_provider="sso_auth", id_user=None):
-    (token, openshift_url) = get_token_and_url()
-    r=None
-    if(exists_openshift_user(token, openshift_url, user_name)):
+
+@application.route("/users/<user_name>", methods=["GET"])
+@auth.login_required
+def get_moc_user(user_name, full_name=None, id_provider="moc-sso", id_user=None):
+    shift = get_openshift()
+    r = None
+    if shift.user_exists(user_name):
         return Response(
             response=json.dumps({"msg": "user (" + user_name + ") exists"}),
             status=200,
-            mimetype='application/json'
-            )
+            mimetype="application/json",
+        )
     return Response(
-            response=json.dumps({"msg": "user (" + user_name + ") does not exist"}),
-            status=400,
-            mimetype='application/json'
-            )       
-
-@application.route("/users/<user_name>", methods=['PUT'])
-def create_moc_user(user_name, full_name=None, id_provider="sso_auth", id_user=None):
-    token, openshift_url = get_token_and_url()
-
-    # A user in OpenShift is composed of 3 parts: user, identity, and identitymapping.
-    user_exists = exists_openshift_user(token, openshift_url, user_name)
-    if not user_exists:
-        r = create_openshift_user(token, openshift_url, user_name, full_name)
-
-        if r.status_code not in [200, 201]:
+        response=json.dumps({"msg": "user (" + user_name + ") does not exist"}),
+        status=400,
```

Comment:
Existence failure should probably return 404 rather than 400.

Reply:
When discussing this with Kristi, we decided on just using 400.

Reply:
Fair enough. We might want to change the status code in `get_moc_rolebindings`, then, where we are returning 404.

Reply:
would it make sense to have a comment there noting the decision to use 400 instead of 404?

Reply:
I don't think so. I only noticed this because of the inconsistency elsewhere, and it is common to not differentiate between 
"user does not exist" and "something else failed" to attempt to prevent someone from mining your user list.

Reply:
Yes, we should be consistent.  I am horribly consistent in my inconsistency.